metadoc-js/src/main/scala/metadoc/MetadocApp.scala

```diff
@@ -21,24 +21,101 @@ object MetadocApp {
       val index = new MutableBrowserIndex(MetadocState(s.Document()))
       registerLanguageExtensions(index)
       val editorService = new MetadocEditorService(index)
-      val input = parseResourceInput(workspace.filenames.head)
-      openEditor(editorService, input)
+
+      def defaultInput = {
+        val input = parseResourceInput(workspace.filenames.head)
+        // Starting with any path, so add the file to the history
+        input.foreach(updateHistory)
+        input
+      }
+
+      /*
+       * Discovering the initial input state may update the history so resolve the
+       * input before registering the history popstate handler to avoid any event
+       * being triggered.
+       */
+      val input =
+        parseResourceInput(Uri.parse(dom.window.location.hash).fragment)
+          .orElse(defaultInput)
+
+      dom.window.onpopstate = { e: dom.PopStateEvent =>
+        val input = Option(e.state.asInstanceOf[IResourceInput]).orElse(
+          parseResourceInput(Uri.parse(dom.window.location.hash).fragment)
+        )
+        input.foreach(openEditor(editorService))
+      }
+
+      dom.window.onresize = { _: dom.Event =>
+        editorService.resize()
+      }
+
+      input.foreach(openEditor(editorService))
     }
   }
 
-  def parseResourceInput(defaultPath: String): IResourceInput = {
-    val path = Option(dom.window.location.hash.stripPrefix("#/"))
+  val SelectionRegex = """L(\d+)(C(\d+))?(-L(\d+)(C(\d+))?)?""".r
+
+  def parseResourceInput(location: String): Option[IResourceInput] = {
```

Comment:
Seems like the "url <-> selection" logic should be in a separate file. wdyt?

Reply:
Yes, I agree. Will clean that up and write some unit tests.